lib/ansible/utils/collection_loader.py

```diff
@@ -47,6 +49,12 @@ def __init__(self, config=None):
         elif self._n_configured_paths is None:
             self._n_configured_paths = []
 
+        # Append all ``ansible_collections`` dirs from sys.path to the end
+        for path in sys.path:
+            sys_installed_path = os.path.join(os.path.abspath(path), 'ansible_collections')
+            if os.path.isdir(sys_installed_path):
+                self._n_configured_paths.append(sys_installed_path)
```

Comment:
This adds paths at the end of the list. Which means that it'll be shadowed by user-provided paths. Is this intended? Should this also be documented somewhere?

Reply:
It is definitely the intent; this feature primarily exists to support ACD and have an "it just works" fallback for 2.9-authored content to transparently resolve to, and depending on how downstream ansible-base ends up being packaged, possibly an RPM-installed version of same. User-installed content should absolutely be able to mask it.

Docs and tests are both missing from this PR at the moment, I'd argue that both need to be addressed before it's merged. That should be achievable once migration is behind us.

Reply:
> Docs and tests are both missing from this PR at the moment

Scratch docs. One of the caveats to allowing this feature, is that no documentation is provided for users about it.  This is purely an internal thing that we are effectively just not communicating about its existence. It will be unsupported for any users who find it and try to use it.  It's basically "for us only".

Reply:
Personally, I'd want to have it supported b/c it's easier to ship dists this way in some cases.

Reply:
> Personally, I'd want to have it supported b/c it's easier to ship dists this way in some cases

That decision is mostly out of my hands. The decision came from "higher" up.  The supported way of distributing collections is galaxy/ah only.

Reply:
Just for the record, I agree with @webknjaz - although I understand it's out of your hands. Being able to pip install a library and have that result in a corresponding ansible collection being installed would be hugely beneficial for some things. Mostly saying that in case further conversations happen by those who are higher up to register user interest in such a feature.

Reply:
@emonty FTR this is how ACD will work @ PyPI @ Ansible 2.10. So it'd be pretty easy for users to figure out how it works. Maybe it's not something that business desires, but the community would definitely want this.